cartoframes/carto_vl/map/map.py

```diff
@@ -0,0 +1,318 @@
+          - if a `dict`, Mapbox or other style as the value of the `style` key.
+            If a Mapbox style, the access token is the value of the `token`
+            key.
+        bounds (dict or list): a dict with `east`,`north`,`west`,`south`
+          properties, or a list of floats in the following order: [west,
+          south, east, north]. If not provided the bounds will be automatically
+          calculated to fit all features.
+
+    Example:
+
+        .. code::
+
+            from cartoframes.carto_vl import carto
+            from cartoframes import CartoContext
+
+            context = CartoContext(
+                base_url='https://your_user_name.carto.com',
+                api_key='your api key'
+            )
+
+            carto.Map([carto.Layer('table in your account')], context)
+
+        CARTO basemap style.
+
+        .. code::
+
+            from cartoframes.carto_vl import carto
+            from cartoframes import CartoContext
+
+            context = CartoContext(
+                base_url='https://your_user_name.carto.com',
+                api_key='your api key'
+            )
+
+            carto.Map(
+                [carto.Layer('table in your account')],
+                context,
+                basemap=vector.Basemaps.darkmatter
+            )
+
+        Custom basemap style. Here we use the Mapbox streets style, which
+        requires an access token.
+
+        .. code::
+
+            from cartoframes.carto_vl import carto
+            from cartoframes import CartoContext
+
+            context = CartoContext(
+                base_url='https://your_user_name.carto.com',
+                api_key='your api key'
+            )
+
+            basemap = {
+                'style': 'mapbox://styles/mapbox/streets-v9',
+                'token: '<your mapbox token>'
+            }
+
+            carto.Map(
+                [carto.Layer('table in your account')],
+                context,
+                basemap
+            )
+
+        Custom bounds
+
+        .. code::
+
+            from cartoframes.carto_vl import carto
+            from cartoframes import CartoContext
+
+            context = CartoContext(
+                base_url='https://your_user_name.carto.com',
+                api_key='your api key'
+            )
+
+            bounds = {
+                'west': -10,
+                'east': 10,
+                'north': -10,
+                'south': 10
+            }
+
+            carto.Map(
+                [carto.Layer('table in your account')],
+                context,
+                bounds
+            )
+    """
+
+    @utils.temp_ignore_warnings
+    def __init__(self,
+                 layers,
+                 context=None,
+                 size=(1024, 632),
+                 basemap=Basemaps.voyager,
+                 bounds=None,
+                 template=None,
+                 **kwargs):
+
+        self.layers = layers
```

Comment:
This was always on my TODO list but I hadn't gotten around to it yet.

Another related issue is this one on how the layers are reverse order, which is unintuitive: https://github.com/CartoDB/cartoframes/issues/555

Reply:
Related to this, we should make it easy to not have a data layer in the map. If `layers=None` or if `layers` is not set at all, we should just give back the basemap (at the zoom/center specified) for a user.

Reply:
Nice! I like this flexibility. Should we also raise an error when there're more than 8 layers?

Reply:
We raised the error for carto.js side because that's where builder stopped. My guess is that performance on an 8 layer carto vl map is slower (depending on the data), but my feeling is that we should give a warning instead that it's best practice to have fewer data layers.